### PR TITLE
Tama compatibility

### PR DESCRIPTION
**Pull request by MarijnS95**
Tama FPC works slightly differently, now requiring a proper IRQ in `wait_finger_lost`. It also requires to loop on `capture_image() = 3`, for some odd reason.
Also, it seems to require much more CPU power than the other platforms, with a usual freeze in `capture_image`. Therefore, request the Power HAL to speed up before claiming a single core at low freq.

Request for testing on all platforms; see if those still work correctly. It has so far been proven okay on Akatsuki and Suzu (albeit it still seems slow and unresponsive on the former, at times - the last commit _kindof_ addresses that). If you have stock experience, please comment on the overal responsiveness and accuracy.

Please do not review this for commit descriptions, dangling comments, typos and formatting yet; that'll all be done **after** confirming these  (if it were on me, I'd clang-format the cr*p out of this before doing anything else - for consistency).

TODO:
- Choose a more suitable powerhint
- Suggestions for removing `USE_FPC_TAMA` from the _main_ `BiometricsFingerprint.cpp` file? Keep in mind that we need to access C++ `IPower`

### CODE REVIEW

**Comment by oshmoun** [comment]
working ok so far
only issue to note is the redundant vibration feedback on failed auth, is it possible the hal is doing two callbacks for the failure case?

**Comment by MarijnS95** [comment]
@oshmoun  Which platform, Tama? No such thing here. Nothing has really changed on the part after the capture. Would you mind providing a logcat with `grep -ie fpc -e finger`?

If you have some time, would you also mind checking out https://github.com/MarijnS95/vendor-sony-oss-fingerprint/tree/tama-fpc? I've been using this branch locally up till last week when finally deciding to refactor that while taking into account `IPower`, but this newer version still feels slower and less responsive. I'll be going over the code again nevertheless.

**Comment by oshmoun** [comment]
yes it's tama, but for some reason the double feedback doesn't happen anymore.
in any case, yeah I can feel some slowness with this PR. Will try the branch later.

**Comment by oshmoun** [comment]
`tama-fpc` is definitely better than this, the speed difference is quite tangible

**Comment by MarijnS95** [comment]
@oshmoun Thanks for confirming that it wasn't just me :). Please try with the latest commit, feels like that fixes it :grin:

**Comment by oshmoun** [comment]
ah, seems I'm a bit late
I can say that the addition of https://github.com/sonyxperiadev/vendor-sony-oss-fingerprint/commit/85382c27ed8457d870e36c301fd946ac37374818 did indeed help before the latest update

**Comment by EnJens** [comment]
I think most cases of USE_FPC_TAMA should be moved into the implementations instead.
E.g. fpc_wake simply becoming a no-op in the implementations that don't use it.

Second: Maybe it's time to put tama in it's own file? It's starting to diverge quite a lot

**Comment by MarijnS95** [comment]
@EnJens I've been thinking about going back to the older `tama-fpc` branch again (and cherry-pick a few cleanups/changes from here), which has the `USE_FPC_TAMA` cases in the platform specific file. It also does away with powerhints, which are not operational yet anyway (since we are probably going to have to add our own that make sense for this case).
The reason to go back is that I have promised "properly" working Tama FPC for months, while keeping these changes "to myself" (and on a branch on my fork). Properly refactoring this all like we've discussed before is going to take a bit more time (lots to do already), and I don't want that to hold back Tama FPC.

In summary:
- Get rid of `captureImage` again, move Tama stuff to the Yoshino/Nile/Tama file (no reason to abstract it for just Tama if a full refactor is around the corner);
- Check fpc_wake again, it really seems like `is_finger_not_on_sensor`;
- Get this reviewed and merged;
- Start working on the refactorings: most likely in C++.

**Comment by MarijnS95** [comment]
@EnJens I've gone ahead and do as mentioned in the last comment; `IPower` is gone. There's no point in separating Tama from Yoshino and Nile for just these two `ifdef`s if we plan to refactor it all anyway in the near future.
Also, the stock hal doesn't use the return value for fpc_wake, which is odd given the values we saw. However, I'll leave it in there as some sort of awake call for now. If I remember correctly, and following the respective commit description, this _did_ make a difference in unlocking times.

Finally, the last commit is an ugly one to account for a race condition. The only way to cleanly solve it is by getting rid of `running_state` (or the `return false` in `setState`), which is only used to validate our HAL and the Service anyway (it is "guaranteed" that the Service is not going to request a new operation without cancelling the running one first). It's existence has proven useful in earlier debugging sessions.

Everyone: This is ready for review, feel free to test it on all platforms and scrutinize the code. Please also compile-test it on 8.1; I have deleted my environment already and am not able to sync it for the time being.
